webapp/src/utils/validationUtils.ts

```diff
@@ -0,0 +1,192 @@
+import { t } from "i18next";
+
+////////////////////////////////////////////////////////////////
+// Types
+////////////////////////////////////////////////////////////////
+
+interface ValidationOptions {
+  existingValues?: string[];
+  excludedValues?: string[];
+  isCaseSensitive?: boolean;
+  allowSpecialChars?: boolean;
+  allowedChars?: string;
+  allowSpaces?: boolean;
+  editedValue?: string;
+  min?: number;
+  max?: number;
+}
+
+////////////////////////////////////////////////////////////////
+// Validators
+////////////////////////////////////////////////////////////////
+
+/**
+ * Validates a single string value against specified criteria.
+ *
+ * Validates the input string against a variety of checks including length restrictions,
+ * character validations, and uniqueness against provided arrays of existing and excluded values.
+ *
+ * @param value - The string to validate. Leading and trailing spaces will be trimmed.
+ * @param options - Configuration options for validation.
+ * @param options.existingValues - An array of strings to check against for duplicates. Comparison is case-insensitive by default.
+ * @param options.excludedValues - An array of strings that the value should not match.
+ * @param options.isCaseSensitive - Whether the comparison with `existingValues` and `excludedValues` is case-sensitive. Defaults to false.
+ * @param options.allowSpecialChars - Flags if special characters are permitted in the value.
+ * @param options.allowedChars - A string representing additional allowed characters outside the typical alphanumeric scope.
+ * @param options.allowSpaces - Flags if spaces are allowed in the value.
+ * @param options.editedValue - The current value being edited, to exclude it from duplicate checks.
+ * @param options.min - Minimum length required for the string. Defaults to 0.
+ * @param options.max - Maximum allowed length for the string. Defaults to 255.
+ * @returns True if validation is successful, or a localized error message if it fails.
+ */
+export function validateString(
+  value: string,
+  options?: ValidationOptions,
+): string | true {
+  const {
+    existingValues = [],
+    excludedValues = [],
+    isCaseSensitive = false,
+    allowSpecialChars = true,
+    allowSpaces = true,
+    allowedChars = "&()_-",
+    editedValue = "",
+    min = 0,
+    max = 255,
+  } = options || {};
+
+  const trimmedValue = value.trim();
+
+  if (!trimmedValue) {
+    return t("form.field.required");
+  }
+
+  if (!allowSpaces && trimmedValue.includes(" ")) {
+    return t("form.field.spacesNotAllowed");
+  }
+
+  if (trimmedValue.length < min) {
+    return t("form.field.minValue", { 0: min });
+  }
+
+  if (trimmedValue.length > max) {
+    return t("form.field.maxValue", { 0: max });
+  }
+
+  // Compiles a regex pattern based on allowed characters and flags.
+  const allowedCharsPattern = new RegExp(
+    generatePattern(allowSpaces, allowSpecialChars, allowedChars),
```

Comment:
`allowSpaces` is checked twice

Reply:
The regex pattern generation (generatePattern function), which conditionally includes spaces in the allowed character set. This does not directly check if the input contains spaces but rather allows them in the validation process if `allowSpaces` is true.

Reply:
e.g. you can have a case were some special chars are needed but spaces forbidden, you need to make sure the validation regex don't allow spaces too
```javascript
 validateString(v, {
                    existingValues: existingCandidates,
                    allowSpaces: false,
                    allowedChars: "&_*",
                  }),
```